lib/rspec/sql.rb

```diff
@@ -28,6 +31,10 @@ module Sql; end
     end
 
     failure_message do |_block|
+      if expected.is_a?(Enumerator) && expected.inspect.match?(/:times>$/)
+        expected = expected.size
+      end
+
```

Comment:
I was thinking today that we should create a new object:
```rb
@matcher = QueryMatcher.new(@queries)
@matcher.matches?(expected)
```

That matcher would implement the different cases of comparing to different types. It would be easier to test that class and by storing it in an ivar, we can re-use it to print the error message.

Reply:
That sounds like a good idea 👍